src/osd/scrubber/scrub_backend.cc

```diff
@@ -1817,21 +1826,68 @@ std::optional<snap_mapper_fix_t> ScrubBackend::scan_object_snaps(
   }
   set<snapid_t> obj_snaps{p->second.begin(), p->second.end()};
 
-  set<snapid_t> cur_snaps;
-  int r = snaps_getter.get_snaps(hoid, &cur_snaps);
-  if (r != 0 && r != -ENOENT) {
-    derr << __func__ << ": get_snaps returned " << cpp_strerror(r) << dendl;
-    ceph_abort();
+  // clang-format off
+
+  // validate both that the mapper contains the correct snaps for the object
+  // and that it is internally consistent.
+  // possible outcomes:
+  //
+  // Error scenarios:
+  // - snapmapper's idea of the object's snaps does not match the object's:
```

Comment:
nit: 
does not match the object:

Reply:
I meant 'the object's idea of its own snapset'

Reply:
How about 'SnapMapper index of object snaps does not match that stored in head object snapset attribute'